### PR TITLE
Add Start Site Detection to Translate DNA Sequences 

### PR DESCRIPTION
**Pull request by KirkD-Glysade**
Tests and resources for Start Site Detection in Translate DNA Sequences

### CODE REVIEW

File: test_df/test_data_functions.py [hunk 1]
```diff
@@ -199,12 +199,42 @@ def test_named_data_function_translate_sequences(self) -> None:
 
     def test_translate_sequences_script(self) -> None:
         from test_df.translate_sequences_script import execute
-        file_in = os.path.join(os.path.dirname(__file__), 'resources', 'translate_sequences_script.json')
-        _, response = run_script(file_in, execute)
+        # file_in = os.path.join(os.path.dirname(__file__), 'resources', 'translate_sequences_script.json')
```
Comment (KirkD-Glysade) [inline test_df/test_data_functions.py]: I'll remove these commented lines once the PR is approved.  I didn't want to completely lose the previous test code.

File: test_df/translate_sequences_script.py [hunk 2]
```diff
@@ -7,13 +7,35 @@
 from df.bio_helper import column_to_sequences, sequences_to_column
```
Comment (KirkD-Glysade) [inline test_df/translate_sequences_script.py:7]: This file duplicates the script found in the Data Function .yaml file.   (Just making sure I've done this correctly.)

Comment (jones-gareth) [reply test_df/translate_sequences_script.py:7]: You might want to look at what Dave Cosgrove has done for the some of his chemistry data functions.  He wrote `test_chem/yaml_to_script.py` which has code to copy the script from DataFxns (assuming that DataFxns and DataFxnPylibTest are in the same directory). 

See `test_chem.test_ArbitraryReaction.ScriptTest.setUp()`

File: test_df/test_data_functions.py (hunk 1 above)
Comment (KirkD-Glysade) [reply test_df/test_data_functions.py]: Removed commented code - can always roll back if needed.

File: test_df/translate_sequences_script.py (hunk 2 above)
Comment (KirkD-Glysade) [reply test_df/translate_sequences_script.py:7]: I used `test_chem/yaml_to_script.py` and verified the content of the `execute` function here matches the results of that script as well as that in the .yaml file. (One vestigial import which came from the original example was removed.)   This test is slightly different than those in `test_chem/`, however, as there is a controlling script, `test_data_functions.py`, which imports the `execute` function from this file.  Both of these files pre-existed in this arrangement, and I modified the content of each to support the new start site detection functionality.  All new tests that I've added as well as the original tests pass.

**Pull request merged**